### PR TITLE
Don't validate github url before downloading in hub

### LINKED ISSUE #61755: HTTP Error 403 for torch.hub ResNet
## 🐛 Bug

`torch.hub` pretrained resnet doesn't work.

## To Reproduce

On colab:

```py
import torch

model = torch.hub.load('pytorch/vision:v0.9.0', 'resnet18', pretrained=True)
```

Error:

```py
/usr/lib/python3.7/urllib/request.py in http_error_default(self, req, fp, code, msg, hdrs)
    647 class HTTPDefaultErrorHandler(BaseHandler):
    648     def http_error_default(self, req, fp, code, msg, hdrs):
--> 649         raise HTTPError(req.full_url, code, msg, hdrs, fp)
    650 
    651 class HTTPRedirectHandler(BaseHandler):

HTTPError: HTTP Error 403: rate limit exceeded
```

This is identical to the code snippet on the [website](https://pytorch.org/hub/pytorch_vision_resnet/)

## Expected behavior

The code should run.

## Environment

Google colab.

cc @nairbv

### PR DESCRIPTION
**Pull request by cyyever**
Fixes #61755

### CODE REVIEW

**Comment by NicolasHug** [comment]
Thanks @cyyever for the fix.

I wonder if torchhub should be hitting the GitHub API  on behalf of the user in the first place? Like, users might want to use the GitHub API locally for other reasons, but if the limit is hit by torchhub, they will be unable to?

**Comment by cyyever** [comment]
> Thanks @cyyever for the fix.
> 
> I wonder if torchhub should be hitting the GitHub API on behalf of the user in the first place? Like, users might want to use the GitHub API locally for other reasons, but if the limit is hit by torchhub, they will be unable to?

Usually it should not, and I will add an option to control this behavior. In fact, if the user gives a wrong repo, it will fail to download the zip.

File: torch/hub.py
Comment (NicolasHug) [inline torch/hub.py:254]: `_get_cache_or_reload` isn't just called from `list` but also from `help` and from `load`.

I wonder if adding  an extra parameter is warranted here.

Perhaps I'm missing a bit of context: Could you describe what https://github.com/pytorch/pytorch/pull/56138 was trying to fix in the first place?

Comment (cyyever) [reply torch/hub.py:254]: _validate_not_a_forked_repo existed before #56138. #56138 fixed Github API it used.

Comment (NicolasHug) [reply torch/hub.py:254]: Thanks,

Do you know what it's trying to fix? If we removed `_validate_not_a_forked_repo`, users would get a 404 instead of an `Exception`. The Exception is a bit better I would agree, but I feel like it's not worth adding a new parameter, or even just using GitHub API credit on behalf of the user without them even knowing. Thoughts?

Comment (cyyever) [reply torch/hub.py:254]: I agree with you. I would rather delete it.

**Review by NicolasHug — APPROVED** [review]
Thanks @cyyever , this LGTM, following our discussion in https://github.com/pytorch/pytorch/pull/61761#discussion_r671262515

We might need to wait for another approval from someone that has more context on this though. Possibly @fmassa or @soumith ?

**Comment by NicolasHug** [comment]
Or perhaps @mthrok can take a look?

**Review by nairbv — APPROVED** [review]
Looks to me like this cleanly reverts: https://github.com/pytorch/pytorch/pull/56138

**Comment by mthrok** [comment]
I do not have the full context, and for hot-fix, this PR looks okay. But if the original reason this validation was introduced still stands, probably we should mock the HTTP requests in tests and I think that's the right fix.

**Comment by jsheng-jian** [comment]
will this fix be included in the 1.9.0 wheel

**Comment by pmeier** [comment]
@jsheng-jian 

> will this fix be included in the 1.9.0 wheel?

It can't because `1.9.0` is already released. I don't know if there will be a `1.9.1` release. Otherwise we need to wait for `1.10.1`.

**Comment by malfet** [comment]
This looks like a revert of https://github.com/pytorch/pytorch/pull/56138, isn't it?

**Comment by NicolasHug** [comment]
As mentioned in https://github.com/pytorch/pytorch/pull/61761#discussion_r671254203 this is also a revert of the original https://github.com/pytorch/pytorch/pull/54451

**Comment by malfet** [comment]
I think the problem with  #56138, is that it introduced infinite loop by incrementing `page` variable outside of the loop body:
https://github.com/pytorch/pytorch/blob/a9b0a921d592b328e7e80a436ef065dadda5f01b/torch/hub.py#L122-L132

Which means this sounds like a more proper fix to the problem: https://github.com/pytorch/pytorch/pull/62072

**Comment by pmeier** [comment]
What is the status on this? `torch.hub` is effectively broken for a lot of people pytorch/vision#4156. This should at least make it into the next release.

**Comment by NicolasHug** [comment]
Sadly, it won't be fixed until the next release :(

I'm not a fan of the solution but #62072 will likely superseed this PR. I opened https://github.com/pytorch/hub/pull/221 to figure out whether #62072 is an appropriate fix or not.

**Comment by NicolasHug** [comment]
According to https://github.com/pytorch/hub/pull/221's CI (https://app.circleci.com/pipelines/github/pytorch/hub/1254/workflows/8d1bd4ce-928b-4532-ac4b-98ae33ea7551/jobs/1687) , the "rate limit exceeded" issue persists.

So we might need to merge this PR instead, until we find a robust solution to validate repos URLs. @malfet WDYT?

**Comment by NicolasHug** [comment]
@malfet perhaps I'm missing something but according to the torchhub CI when using pytorch nightly ( pytorch/hub#221), where both #62072 and #62139 are in use, these rate API issues persist. So I'm afraid that these fixes might not be enough.

**Comment by malfet** [comment]
@NicolasHug can you please share a link to CI log are referring to?

**Comment by NicolasHug** [comment]
Sure, it's just the CI at the bottom of https://github.com/pytorch/hub/pull/221 : 

https://app.circleci.com/pipelines/github/pytorch/hub/1254/workflows/8d1bd4ce-928b-4532-ac4b-98ae33ea7551/jobs/1687

**Comment by malfet** [comment]
@NicolasHug your PR failed to move CI from 1.9 to nightly, I've committed proposed [change](https://github.com/pytorch/hub/pull/221/commits/22cd927d32453b90a1183a4b5918c3f4ea7a754c) to that PR and after that CI is green, see https://app.circleci.com/pipelines/github/pytorch/hub/1264/workflows/6ec0e536-94f5-4414-b905-1d61bff8f6ae/jobs/1698

**Comment by NicolasHug** [comment]
Oh I see, thanks a lot @malfet for taking a look. Since the issue is fixed on pytorch master I think it's safe to close this PR. Thanks for the original work @cyyever !

**Pull request closed without merging**
